src/commands/createFunction/CSharpFunctionCreator.ts

```diff
@@ -39,8 +39,9 @@ export class CSharpFunctionCreator extends FunctionCreatorBase {
             this._functionName = functionName;
         }
 
-        if (functionSettings.length > 0) {
-            this._namespace = <string>functionSettings.shift();
+        const namespaceKey: string = 'namespace';
```

Comment:
Is empty string a valid setting value for namespace?

Reply:
It is not valid. However, now that you mention it I'm torn on what we should do here. I'm leaning towards we should pass along the empty string (which will eventually result in an exception). I'm not sure we want to prompt for the namespace when we're being called as an api like this